torch/nested/_internal/nested_tensor.py

```diff
@@ -83,6 +102,45 @@ def __repr__(self):
             grad_fn_str = f", grad_fn={self.grad_fn}"
         return f"NestedTensor(size={self._size}, offsets={self.offsets}{grad_fn_str})"
 
+    def __tensor_flatten__(self):
+        return ["_values", "_offsets"], (self.requires_grad, self.ragged_size)
+
+    @staticmethod
+    def __tensor_unflatten__(inner_tensors: Dict, meta):
+        assert len(inner_tensors) == 2
+        values = inner_tensors["_values"]
+        offsets = inner_tensors["_offsets"]
```

Comment:
do we get the exact same `offsets` and `values` tensors after a cycle of `__tensor_flatten__()` -> `__tensor_unflatten__()` ?

Reply:
It would depend on what kind of function you pass to `transform_subclass` (which is usually when the flatten/unflatten cycle happens). During fakification, for example, the fakification callback is passed, and so the we would get the fakified versions of offsets and values back.